manifests/service.pp

```diff
@@ -12,9 +12,12 @@
   $service_enable  = $nsclient::service_enable,
   $allowed_hosts   = $nsclient::allowed_hosts,
   $config_template = $nsclient::config_template,
-  $install_path    = $nsclient::install_path
+  $install_path    = $nsclient::install_path,
+  $password        = $nsclient::password
```

Comment:
the tests itself were okay. The variable `$password` was just unknown within the scope of the template. The template is called in the class nsclient::service, so the variable has to exist here too. I added it as a new class parameter.